### PR TITLE
Add competition based on PP in multiplayer games.

### PR DESCRIPTION
**Pull request by RealistikDash**
Use `!mp pp` to toggle it on and off (off by default).

### CODE REVIEW

**Comment by 7ez** [comment]
Performance Points***

File: peppy/adapters/performance_service.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        beatmap_id: int,
+        mode: int,
+        mods: int,
+        max_combo: int,
+        accuracy: float,
+        miss_count: int,
+        passed_objects: Optional[int] = None,
+    ) -> dict[str, Any]:
+        respone = requests.post(
+            self._base_url + "/api/v1/calulcate",
```
Comment (7ez) [inline peppy/adapters/performance_service.py]: f"" 🤓🤓🤓

Comment (tsunyoku) [inline peppy/adapters/performance_service.py]:
```suggestion
            self._base_url + "/api/v1/calculate",
```

**Review by 7ez — CHANGES_REQUESTED** [review]

File: peppy/adapters/performance_service.py (hunk 1 above)
Comment (RealistikDash) [reply peppy/adapters/performance_service.py]: there is a bigger issue.

File: peppy/adapters/performance_service.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                "mods": mods,
+                "max_combo": max_combo,
+                "accuracy": accuracy,
+                "miss_count": miss_count,
+                "passed_objects": passed_objects,
+            },
+            timeout=self._timeout,
+        )
+        respone.raise_for_status()
+        return respone.json()
```
Comment (tsunyoku) [inline peppy/adapters/performance_service.py:60]: response is `list[dict[str, Any]]` (https://github.com/RealistikOsu/performance-service/blob/17a9561e5877f32d249ea4d2d2204df40f155e50/src/api/routes/calculate.rs#L199)

File: peppy/adapters/performance_service.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            self._base_url + "/api/v1/calulcate",
+            data={
+                "beatmap_id": beatmap_id,
+                "mode": mode,
+                "mods": mods,
+                "max_combo": max_combo,
+                "accuracy": accuracy,
+                "miss_count": miss_count,
+                "passed_objects": passed_objects,
+            },
```
Comment (tsunyoku) [inline peppy/adapters/performance_service.py]: data should be an array (https://github.com/RealistikOsu/performance-service/blob/17a9561e5877f32d249ea4d2d2204df40f155e50/src/api/routes/calculate.rs#L198)

**Review by tsunyoku — CHANGES_REQUESTED** [review]

File: peppy/adapters/performance_service.py (hunk 2 above)
Comment (7ez) [reply peppy/adapters/performance_service.py:60]: you mean respone

**Pull request merged**